src/cecs277TermProject/RoachMotel.java

```diff
@@ -52,6 +51,19 @@ public void checkOut(RoachColony roachColony) {
 		availableRooms.add(occupiedRooms.remove(roachColony));
 
 		// TODO: [Tyler] Implement payment
+		// I feel like payment and logging methods should be called outside of here, otherwise we need add some member variables to these in order to store
+		// which payment methods go with which colony.  I'm thinking something like the Strategy Lab we did.
+		
+	}
+	
+	/**
+	 * Will allow a colony to pay for the room upon checkout.  A payment method (RoachPal or MasterRoach) will be passed in through the main method.
+	 * @param paymentMethod  A payment method (RoachPal or MasterRoach) passed in through the main method.
+	 * @param cost The total cost to be paid.
+	 */
+	public void pay(PaymentStrategy paymentMethod, double cost)
```

Comment:
This should be private and used only in `checkout()` where the params will be: colony, numberOfDays, and payment method.

Reply:
Do you want me to calculate the cost in that method?

Reply:
yes.

Reply:
do we have a way to tell what room a colony is staying in yet?